distributed/scheduler.py

```diff
@@ -525,16 +534,17 @@ class WorkerState:
 
     def __init__(
         self,
-        address: str = None,
-        pid: Py_ssize_t = 0,
-        name: object = None,
+        *,
+        address: str,
+        pid: Py_ssize_t,
+        name: object,
         nthreads: Py_ssize_t = 0,
-        memory_limit: Py_ssize_t = 0,
-        local_directory: str = None,
-        services: dict = None,
-        versions: dict = None,
-        nanny: str = None,
-        extra: dict = None,
+        memory_limit: Py_ssize_t,
+        local_directory: str,
+        nanny: str,
+        services: "dict | None" = None,
+        versions: "dict | None" = None,
+        extra: "dict | None" = None,
```

Comment:
What does the Cython code around the usage of these variables look like? In the past I've seen Cython handle these `""` types as just `object`s. If that's still true, that would mean Cython does not use the faster C level APIs associated with these objects, which would lose some of the performance gains we got with Cythonizing the scheduler.

Reply:
You're right, they degrade performance. Reverted.